internal/provider/validation.go

```diff
@@ -36,22 +38,41 @@ func validateAccount(apiClient *apiClient, sub subi.SubstrateExt) error {
 	return nil
 }
 
-func validateRMBProxyServer(apiClient *apiClient) error {
-	return apiClient.grid_client.Ping()
+func validateRMBProxyServer(threefoldPluginClient *threefoldPluginClient) error {
+	return threefoldPluginClient.gridProxyClient.Ping()
 }
 
-func validateRMBProxy(apiClient *apiClient) error {
-	if err := validateRMBProxyServer(apiClient); err != nil {
-		return err
+func validateMnemonics(mnemonics string) bool {
+	return bip39.IsMnemonicValid(mnemonics)
+}
+
+func validateSubstrateURL(url string) error {
+	if len(url) == 0 {
+		return errors.New("substrate url is required")
 	}
+
+	alphaOnly := regexp.MustCompile(`^wss:\/\/[a-z0-9]+\.[a-z0-9]\/?([^\s<>\#%"\,\{\}\\|\\\^\[\]]+)?$`)
+	if !alphaOnly.MatchString(url) {
+		return errors.New("substrate url is not valid")
+	}
+
 	return nil
 }
 
-func preValidate(apiClient *apiClient, sub subi.SubstrateExt) error {
-	return validateRMBProxy(apiClient)
+func validateProxyURL(url string) error {
```

Comment:
do we still need that for the relay? also the relay works on wss no?

Reply:
this validate the gridproxy's url, which is still used even after introducing the relay stuff